pkg/apis/core/validation/validation.go

```diff
@@ -4283,6 +4289,76 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	return allErrs
 }
 
+func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	if spec.Resources == nil {
+		return nil
+	}
+
+	allErrs := field.ErrorList{}
+	resourceOpts := ResourceValidationOptions{
+		AllowIndivisibleHugePagesValues: opts.AllowIndivisibleHugePagesValues,
+		PodLevelResource:                true,
+	}
+
+	// ValidateResourceRequirements checks if resource names and quantities are
+	// valid, and requests are less than limits.
+	allErrs = append(allErrs, ValidateResourceRequirements(spec.Resources, podClaimNames, fldPath, resourceOpts)...)
+	allErrs = append(allErrs, validatePodResourceConsistency(spec, fldPath)...)
+	return allErrs
+}
+
+// validatePodResourceConsistency checks if effective container-level requests are
+// less than or equal to pod-level requests, and individual container-level limits
+// are less than or equal to pod-level limits.
+func validatePodResourceConsistency(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	v1PodSpec := &v1.PodSpec{}
+	if err := corev1.Convert_core_PodSpec_To_v1_PodSpec(spec, v1PodSpec, nil); err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath, fmt.Errorf("invalid %q: %v", fldPath, err.Error())))
+	}
+
+	reqPath := fldPath.Child("requests")
+	// resourcehelper.PodRequests method requires a Pod object to calculate the total
+	// requests requirements of a pod. Hence a local copy of Pod object is created from PodSpec to
+	// facilitate proper resource calculation without modifying PodRequests method.
+	podReqsFromContainers := resourcehelper.PodRequests(&v1.Pod{Spec: *v1PodSpec}, resourcehelper.PodResourcesOptions{
+		InPlacePodVerticalScalingEnabled: utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling),
+		PodLevelResourcesEnabled:         utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources),
+		ExcludeOverhead:                  true,
+		UseContainerLevelResources:       true,
+	})
+
+	// Pod-level requests >= effective requests of all containers in a pod.
+	for resourceName, quantity := range podReqsFromContainers {
+		podSpecRequests, exists := spec.Resources.Requests[core.ResourceName(resourceName.String())]
+		if !exists {
+			continue
+		}
+
+		fldPath := reqPath.Key(resourceName.String())
+		if quantity.Cmp(podSpecRequests) > 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath, podSpecRequests.String(), fmt.Sprintf("must be greater than or equal to effective %s container requests of %s", resourceName, quantity.String())))
```

Comment:
What does effective container request mean? It does not appear to be a terminology we use in our docs.

Reply:
@ndixita may be we can add some comment to explain the meaning. This is not in the user documentation because it's used internally to decide how much resources is needed for the pod.

Reply:
Ah...I guess we'll have to document this more clearly since the error is user-facing. How about "aggregate container requests at each stage (init or non-init)".

Reply:
I think defining "aggregage container requests" makes sense - it's a better term than "effective" here, IMO.

Reply:
modified, PTAL.